scipy/stats/_axis_nan_policy.py

```diff
@@ -41,34 +120,15 @@ def _broadcast_shapes_remove_axis(shapes, axis=None):
     Same as _broadcast_array_shapes, but given a sequence
     of array shapes `shapes` instead of the arrays themselves.
     """
-    n_dims = max([len(shape) for shape in shapes])
-    new_shapes = np.ones((len(shapes), n_dims), dtype=int)
-    for row, shape in zip(new_shapes, shapes):
-        row[len(row)-len(shape):] = shape  # can't use negative indices (-0:)
-    if axis is not None:
-        new_shapes = np.delete(new_shapes, axis, axis=1)
-    new_shape = np.max(new_shapes, axis=0)
-    new_shape *= new_shapes.all(axis=0)
-    if np.any(~((new_shapes == 1) | (new_shapes == new_shape))):
-        raise ValueError("Array shapes are incompatible for broadcasting.")
-    return tuple(new_shape)
+    shapes = _broadcast_shapes(shapes, axis)
+    shape = np.delete(shapes[0], axis)
```

Comment:
wouldn't work when `axis=None`:

```python
In [1]: from from scipy.stats._axis_nan_policy import _broadcast_shapes_remove_axis

In [2]: _broadcast_shapes_remove_axis([(3, 9, 2, 5), (3, 9, 2, 5)])
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-35-66e4d08b298f> in <module>
----> 1 _broadcast_shapes_remove_axis([(3, 9, 2, 5), (3, 9, 2, 5)])

~/Desktop/scipy_source/scipy/stats/_axis_nan_policy.py in _broadcast_shapes_remove_axis(shapes, axis)
    122     """
    123     shapes = _broadcast_shapes(shapes, axis)
--> 124     shape = np.delete(shapes[0], axis)
    125     return tuple(shape)
    126 

<__array_function__ internals> in delete(*args, **kwargs)

~/Desktop/scipy_source/scipy-dev/lib/python3.9/site-packages/numpy/lib/function_base.py in delete(arr, obj, axis)
   4550         else:
   4551             keep = ones(N, dtype=bool)
-> 4552             keep[obj,] = False
   4553 
   4554         slobj[axis] = keep

IndexError: arrays used as indices must be of integer (or boolean) type
```

I guess we never use this function unless we want to remove _some_ axes. In that case, it's better to not default `axis` to `None` so the signature is just `_broadcast_shapes_remove_axis(shapes, axis)`

Reply:
The previous version did work with `axis=None`. I guess we should preserve that behavior:

```python
def _broadcast_shapes_remove_axis(shapes, axis=None):
    """
    Broadcast shapes, dropping specified axes
    Same as _broadcast_array_shapes, but given a sequence
    of array shapes `shapes` instead of the arrays themselves.
    """
    shapes = _broadcast_shapes(shapes, axis)
    if axis is not None:
        shapes = shapes[0]
    shape = np.delete(shapes, axis)
    return tuple(shape)
```

Reply:
Thanks. I'll change these things when we get CI back.

Reply:
Done! Er, I think you meant:
```python3
    shapes = _broadcast_shapes(shapes, axis)
    shape = shapes[0]
    if axis is not None:
        shape = np.delete(shape, axis)
    return tuple(shape)
```
That's what I did.

Reply:
Yep, you are right! Although, this would still fail for cases like these:

```py
In [1]: import numpy as np

In [2]: from scipy.stats._axis_nan_policy import _check_empty_inputs

In [3]: _check_empty_inputs([np.array([]), np.array([])], None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-8945ea70ea1a> in <module>
----> 1 _check_empty_inputs([np.array([]), np.array([])], None)

~/Desktop/scipy_source/scipy/stats/_axis_nan_policy.py in _check_empty_inputs(samples, axis)
    222     # otherwise, the statistic and p-value will be either empty arrays or
    223     # arrays with NaNs. Produce the appropriate array and return it.
--> 224     output_shape = _broadcast_array_shapes_remove_axis(samples, axis)
    225     output = np.ones(output_shape) * np.nan
    226     return output

~/Desktop/scipy_source/scipy/stats/_axis_nan_policy.py in _broadcast_array_shapes_remove_axis(arrays, axis)
    111     # ravel arrays before broadcasting.
    112     shapes = [arr.shape for arr in arrays]
--> 113     return _broadcast_shapes_remove_axis(shapes, axis)
    114 
    115 

~/Desktop/scipy_source/scipy/stats/_axis_nan_policy.py in _broadcast_shapes_remove_axis(shapes, axis)
    125     if axis is not None:
    126         shape = np.delete(shape, axis)
--> 127     return tuple(shape)
    128 
    129 

TypeError: 'numpy.int64' object is not iterable
```

But from the `_axis_nan_policy_factory` code, it looks like the case of `axis=None` is never hit. So, I think we don't need to worry about this.